src/plugins/rule/schema.js

```diff
@@ -0,0 +1,143 @@
+        }
+        // A non-empty rule without a statement should get an empty statement.
+        if (index === 0) {
+            const statement = Block.create({
+                type: 'rule_statement',
+                nodes: [
+                    // An empty leaf will be added by slate in a subsequent
+                    // normalization.
+                    Block.create({ type: 'paragraph' }),
+                ],
+            })
+            editor.insertNodeByKey(node.key, 0, statement)
+            return
+        }
+        // Otherwise (invalid child after the statement but not at the end of
+        // rule) we fold the invalid child into previous node.
+        const previous = node.nodes.get(index - 1)
+        editor.moveNodeByKey(child.key, previous.key, previous.nodes.size)
+        break
+    }
+
+    case 'child_min_invalid':
+        // A non-empty rule without a statement should get an empty statement.
+        if (index === 0) {
+            const statement = Block.create({
+                type: 'rule_statement',
+                nodes: [
+                    // An empty leaf will be added by slate in a subsequent
+                    // normalization.
+                    Block.create({ type: 'paragraph' }),
+                ],
+            })
+            editor.insertNodeByKey(node.key, 0, statement)
+            return
+        }
+
+        console.warn('Unhandled rule violation:', code)
+        break
+
+    case 'node_data_invalid':
+        // Rule type is either missing or invalid. Replace it with default
+        // ('rule').
+        if (key === 'type') {
+            editor.setNodeByKey(
+                node.key, { data: node.data.set('type', 'rule') })
+            return
+        }
+
+        /* istanbul ignore next */
+        console.warn('Unhandled rule violation: node_data_invalid', key)
+        break
+
+    // Rule was inserted into an invalid parent.
+    case 'parent_object_invalid':
+    case 'parent_type_invalid':
+        editor.unwrapBlockByKey(node.key)
+        break
+
+    /* istanbul ignore next */
+    default:
+        console.warn('Unhandled rule violation:', code)
+        break
+    }
+}
+
+function normalizeContent(editor, error) {
+    const { code, node } = error
+
+    switch (code) {
+    // Statements, proofs, and examples make no sense outside a rule. Should it
+    // happen however, just replace it with its contents.
+    case 'parent_type_invalid':
+        editor.unwrapBlockByKey(node.key)
+        break
+
+    /* istanbul ignore next */
+    default:
+        console.warn('Unhandled rule content violation:', code)
+        break
+    }
+}
+
+export const RULE_PARENT = ['document', 'section']
+
+export default function schema(options) {
+    const content = {
+        match: options.content.map(type => ({ type })),
+        min: 1,
+    }
+
+    return {
+        blocks: {
+            rule: {
+                parent: [{ object: 'document' }, { type: 'section' }],
```

Comment:
If we want to restrict parents then it should be configurable by options

Reply:
Then we probably should also change other plugins (e.g. exercise) which also restrict their parents. For now I'd leave it as is; we're about to start migration to Slate 0.5X which would require schema to be largely rewritten, so I'd prefer not to invest too much time in this old schema.

Reply:
Ok, this argument is good enough for me :D